### PR TITLE
Updated the samples to support the different environment types. Simil…

### PR DESCRIPTION
**Pull request by Teodelas**
Changed the samples and the x-core module to support the new environment types as an optional variable.

- [ X] I have run all the tests locally and they all pass.
- [ X] I have followed the relevant style guide for my changes.

### CODE REVIEW

**Comment by danistrebel** [comment]
@Teodelas Thanks a lot. This is a commonly asked for feature!

Could you please rebase to get the broader bump to CFF 28.0.0 into your branch? Otherwise this looks good at first sight. 

Automated checks still pending.

**Comment by Teodelas** [comment]
I think I did a rebase correctly, but honestly not sure. I don't use git often...

File: samples/x-multi-region/x-demo.tfvars [hunk 1]
```diff
@@ -38,15 +38,18 @@ apigee_environments = {
...
   test2 = {
     display_name = "Test 2"
     description  = "Environment created by apigee/terraform-modules"
     node_config  = null
     iam          = null
     envgroups    = ["test"]
+    type         = null
   }
 }
+}
```
Comment (danistrebel) [inline samples/x-multi-region/x-demo.tfvars:52]: I think this broke the test. Can you remove this extra '}'

**Review by danistrebel — CHANGES_REQUESTED** [review]

File: samples/x-multi-region/x-demo.tfvars (hunk 1 above)
Comment (Teodelas) [reply samples/x-multi-region/x-demo.tfvars]: done.

**Review by danistrebel — APPROVED** [review]
Thanks @Teodelas

**Pull request merged**